internal/sysfs/stat_windows.go

```diff
@@ -50,17 +50,19 @@ func statPath(createFileAttrs uint32, path string) (fsapi.Stat_t, syscall.Errno)
 	return statHandle(h)
 }
 
-func statFile(f *os.File) (fsapi.Stat_t, syscall.Errno) {
-	// Attempt to get the stat by handle, which works for normal files
-	st, err := statHandle(syscall.Handle(f.Fd()))
-
-	// ERROR_INVALID_HANDLE happens before Go 1.20. Don't fail as we only
-	// use that approach to fill in inode data, which is not critical.
-	//
-	// Note: statHandle uses UnwrapOSError which coerces
-	// ERROR_INVALID_HANDLE to EBADF.
-	if err != syscall.EBADF {
-		return st, err
+func statFile(f fs.File) (fsapi.Stat_t, syscall.Errno) {
+	if osF, ok := f.(*os.File); ok {
```

Comment:
this was the only case we actually used os.File, so I special cased it as it is windows only